File: backend-py/send_barcode.py
from backend.req.post_barcode import post_barcode

def send_barcode(barcode, user, jwt, lager_user_ids):
    barcode_c = barcode

    if not jwt:
        notification.notify(
            title="Barcode Scanner",
            message=f"{barcode_c} lokal gespeichert"
        )
        return

    try:
        post_barcode(barcode, user, jwt, lager_user_ids, False)
        notification.notify(
            title="Barcode Scanner",
            message=f"{barcode_c} gesendet"
        )
    except Exception as e:
        print(f"Error: {e}")
        # Hier sollte eine Funktion zum Anzeigen eines Dialogs hinzugefügt werden, z.B.:
        # dialog.alert_default(str(e))

File: src-tauri/src/sqlite/src/send_barcode.rs
use req::write_barcode::write_barcode;

pub fn send_barcode(barcode: String, user: i32, jwt: &str, lager_user_ids: &Vec<i32>) {
    if jwt.is_empty() {
        // Notification::new()
        // .summary(&format!("Barcode Scanner: {} lokal gespeichert", barcode_c))
        // .show()
        // .unwrap();
        // return;
    }

    match write_barcode(barcode, user, jwt, &lager_user_ids, false) {
        Ok(_) => {
        }
        Err(e) => {
            eprintln!("Fehler beim Senden des Barcodes: {}", e);
        }
    }
}